radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/CandidateForkConfig.java

```diff
@@ -64,59 +64,35 @@
 
 package com.radixdlt.statecomputer.forks;
 
-import com.google.common.hash.HashCode;
+import com.google.common.collect.ImmutableSet;
 import com.radixdlt.engine.PostProcessor;
 import com.radixdlt.statecomputer.LedgerAndBFTProof;
 
-public final class CandidateForkConfig implements ForkConfig {
-  private final String name;
-  private final HashCode hash;
-  private final RERules reRules;
-  private int requiredStake;
-  private long minEpoch;
+public record CandidateForkConfig(
+    String name,
+    RERules engineRules,
+    ImmutableSet<Threshold> thresholds,
+    long minEpoch,
+    long maxEpoch)
+    implements ForkConfig {
 
-  public CandidateForkConfig(
-      String name, HashCode hash, RERules reRules, int requiredStake, long minEpoch) {
-    this.name = name;
-    this.hash = hash;
-    this.reRules = reRules;
-    this.requiredStake = requiredStake;
-    this.minEpoch = minEpoch;
+  public static record Threshold(short requiredStake, int numEpochsBeforeEnacted) {
```

Comment:
should we add validation for `requiredStake`?

Reply:
good point, done